sdktests/server_side_data_system_persistence_store.go

```diff
@@ -0,0 +1,35 @@
+package sdktests
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/redis/go-redis/v9"
+)
+
+type PersistenceStore interface {
+	DSN() string
+
+	WriteData(key string, data map[string]string) error
+
+	Reset() error
+}
+
+type RedisPersistenceStore struct {
+	redis *redis.Client
+}
+
+func (r RedisPersistenceStore) DSN() string {
+	return fmt.Sprintf("redis://%s", r.redis.Options().Addr)
+}
+
+func (r RedisPersistenceStore) Reset() error {
```

Comment:
Since these methods are mutating, I think it'd be slightly clearer to have them operate on pointer receivers.

Reply:
Done.